doc/manual/interfaces.rst

```diff
@@ -173,6 +173,8 @@ A key part in defining an ``AbstractArray`` subtype is :func:`Base.linearindexin
 
 This distinction determines which scalar indexing methods the type must define. ``LinearFast()`` arrays are simple: just define :func:`getindex(A::ArrayType, i::Int) <getindex>`.  When the array is subsequently indexed with a multidimensional set of indices, the fallback :func:`getindex(A::AbstractArray, I...)` efficiently converts the indices into one linear index and then calls the above method. ``LinearSlow()`` arrays, on the other hand, require methods to be defined for each supported dimensionality with ``ndims(A)`` ``Int`` indices.  For example, the builtin ``SparseMatrix`` type only supports two dimensions, so it just defines :func:`getindex(A::SparseMatrix, i::Int, j::Int)`.  The same holds for :func:`setindex!`.
 
+The result of indexing an AbstractArray can itself be an array (for instance when indexing by a ``Range``). The AbstractArray fallback methods use :func:`similar` to create an ``Array`` of the appropriate size and element type, which is filled in using the basic indexing method described above. When implementing an array wrapper you often want the result to be wrapped as well, which can be accomplished by defining ``Base.similar{T}(A::YourArrayType, ::Type{T}, dims::Dims)`` to create the appropriate wrapped array.
```

Comment:
Minor nit: I'd say "allocate an ``Array``" rather than "create an ``Array``" just to be precise. But that's a pretty pedantic suggestion.